src/ui/v_configuration.py

```diff
@@ -16,15 +16,34 @@ def update_warning_task_enabled(should_enable):
         call warnings_code.update_warning_check_task_status({should_enable})
     """).collect()
 
+def generate_warnings_once():
+    session.sql(f"""
+        call warnings_code.check_warnings()
+    """).collect()
+
 def render():
     st.header("Configuration")
+    st.caption("""
+        When new sensor data is created, it needs to be scanned for values
+        that are out-of-bounds. A warning will be created for each discovered
+        reading that does not fall within the manufacturer's specified range.
+        Note that enabling the warning generation task can consume a lot of
+        credits in your Snowflake account; the warning generation logic can be
+        manually triggered once by clicking the button below instead.
+    """)
+
+    # one-time warning generation
+    st.button("Generate warnings once (takes a while)", on_click=generate_warnings_once)
+
+    # warning task
     is_warning_task_enabled = check_warning_task_enabled()
-    col1, col2 = st.columns([0.5, 0.5])
-    col1.caption("Enable warning generation task")
-    with col2:
-        new_value = st.checkbox(label='', value=is_warning_task_enabled, key='warning_task_enabled')
-        if is_warning_task_enabled != new_value:
-            update_warning_task_enabled(new_value)
+    new_value = st.checkbox(
```

Comment:
Just to make sure I understand here. We have a button for on-demand warning generation (i.e., generate warnings once), and a checkbox for continuous warning generation. What's the behavior when the checkbox = true and the button is clicked? Am I thinking about this incorrectly?

Reply:
The button calls the same procedure the task calls. So clicking the button would in effect be hastening the run of the next task execution. I don't think there's any data integrity danger, but we can also disable the button if the checkbox is selected if that makes more sense to you.

Reply:
I think that makes sense. What about the other way around? I see the note "Generate warnings once (takes a while) - should the checkbox be enabled (i.e., able to be clicked) if the warning generation task (kicked off by clicking the button) is running? I could see it becoming temporarily disabled until the task is complete. Just my .02.

Reply:
Makes sense -- and in fact the streamlit already pauses until the function returns, so there's no risk of that happening anyway. I'm making the change to prevent the button from being clicked if the task is enabled.